torch/_dynamo/output_graph.py

```diff
@@ -249,6 +250,8 @@ def __init__(
             )
         )
 
+        self.guards.add(GradModeSource().make_guard(GuardBuilder.GRAD_MODE))
+
```

Comment:
Out of curiosity, is the sole purpose of this guard for logging purposes?

The actual guard logic exists in the GlobalStateGuard which is opaque to how we register guard metadata, right?